src/components/PopupWithForms.js

```diff
@@ -1,14 +1,23 @@
 import Popup from "./Popup.js";
 
 export default class PopupWithForm extends Popup {
-  constructor(popupSelector, handleFormSubmit, formSelector) {
+  constructor(popupSelector, handleFormSubmit, settings) {
     super({ popupSelector });
     this._handleFormSubmit = handleFormSubmit;
-    this._popupForm = this._popupElement.querySelector(formSelector);
+    this._button = this._popupElement.querySelector(
+      settings.submitButtonSelector
+    );
+    this._buttonText = this._button.textContent;
+    this._popupForm = this._popupElement.querySelector(settings.formSelector);
     this._inputList = Array.from(
       this._popupForm.querySelectorAll(".modal__input")
     );
-    this._submitButton = this._popupForm.querySelector(".modal__button");
+  }
+
+  renderSaving(isSaving) {
+    isSaving
+      ? (this._button.textContent = "Saving...")
+      : (this._button.textContent = this._buttonText);
   }
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`